### PR TITLE
Add autoconf to the management interface in IPv6

### PR DESCRIPTION
**Pull request by Nambrok**
Allows to configure the management interface in Autoconf when the interface primary type is IPv6.

### CODE REVIEW

**Comment by Nambrok** [comment]
@benjamreis

File: plugins-base/XSFeatureInterface.py [hunk 1]
```diff
@@ -175,8 +177,10 @@ def UpdateFieldsPRECOMMIT(self):
                 pane.AddStatusField(Lang("Netmask",  16),  self.netmask)
                 pane.AddStatusField(Lang("Gateway",  16),  self.gateway)
 
-            if self.mode != 'Static' and self.hostname == '':
+            if self.mode == 'DHCP' and self.hostname == '':
                 pane.AddStatusField(Lang("Hostname",  16), Lang("Assigned by DHCP"))
+            elif self.mode == 'Autoconf' and self.hostname == '':
+                pane.AddStatusField(Lang("Hostname", 16), Lang("Automatically assigned"))
```
Comment (benjamreis) [inline plugins-base/XSFeatureInterface.py:184]: I think both case could be in the same If with a generic sentence about auto assignation.

Comment (Nambrok) [reply plugins-base/XSFeatureInterface.py:184]: I think "Assigned by DHCP" has a specific meaning. Which does not apply to Autoconf. Making it not appear in Autoconf might be a better solution?

File: plugins-base/XSFeatureInterface.py [hunk 2]
```diff
@@ -53,11 +53,13 @@ def __init__(self):
...
-        self.modeMenu = Menu(self, None, Lang("Select IP Address Configuration Mode"), [
-            ChoiceDef(Lang("DHCP"), lambda: self.HandleModeChoice('DHCP2') ),
-            ChoiceDef(Lang("DHCP with Manually Assigned Hostname"), lambda: self.HandleModeChoice('DHCPMANUAL') ),
-            ChoiceDef(Lang("Static"), lambda: self.HandleModeChoice('STATIC') )
-            ])
+        mode_choicedefs = []
+        if(currentPIF['primary_address_type'].lower() == 'ipv6'):
+            mode_choicedefs.append(ChoiceDef(Lang("Autoconf"), lambda : self.HandleModeChoice("AUTOCONF") ))
+        mode_choicedefs.append(ChoiceDef(Lang("DHCP"), lambda: self.HandleModeChoice('DHCP2') ))
+        mode_choicedefs.append(ChoiceDef(Lang("DHCP with Manually Assigned Hostname"), lambda: self.HandleModeChoice('DHCPMANUAL') ))
```
Comment (stormi) [inline plugins-base/XSFeatureInterface.py]: I don't know if there are strict coding style rules on xsconsole, but we should at least try to remain under 120 chars per line.

**Review by stormi — APPROVED** [review]
It looks good to me. One minor comment on formatting.

I let you decide with Benjamin what's best for the hostname question.

Regarding the feature, I don't see added code to handle the Autoconf choice once chosen by the user. Is setting `self.mode` all we have to do for it to be effective?

**Comment by benjamreis** [comment]
> Regarding the feature, I don't see added code to handle the Autoconf choice once chosen by the user. Is setting `self.mode` all we have to do for it to be effective?

Yes as the reconfigure call are done with self.mode as an arg :)

**Review by benjamreis — APPROVED** [review]

**Comment by benjamreis** [comment]
Upstream contribution included in https://github.com/xapi-project/xsconsole/pull/4

**Pull request merged**